code/datums/tutorial/marine/reqs_line.dm

```diff
@@ -0,0 +1,470 @@
+			update_objective("Get the M41 Extended magazine and perform a high toss to give it to the forgetful marine.")
+
+/datum/tutorial/marine/reqs_line/start_tutorial(mob/starting_mob)
+	. = ..()
+	if(!.)
+		return
+
+	init_mob()
+	START_PROCESSING(SSfastobj, src)
+
+	message_to_player("Welcome to Requisitions Line tutorial. Come in and have a seat.")
+	update_objective("Reach the line window to begin!")
+	var/turf/target_turf = loc_from_corner(3, 6)
+	RegisterSignal(target_turf, COMSIG_TURF_ENTERED, PROC_REF(user_in_position))
+
+
+/// Called when the player is in position to start handling the line
+/datum/tutorial/marine/reqs_line/proc/user_in_position(turf/source, atom/movable/mover)
+	SIGNAL_HANDLER
+	if(mover != tutorial_mob)
+		return
+	UnregisterSignal(source, COMSIG_TURF_ENTERED)
+	stage = TUTORIAL_REQS_LINE_STAGE_ATTACHIES
+	var/list/request = list(/obj/item/attachable/magnetic_harness, /obj/item/attachable/extended_barrel)
+	spawn_agent(request)
+
+/datum/tutorial/marine/reqs_line/proc/continue_stage_gearbox()
+	TUTORIAL_ATOM_FROM_TRACKING(/obj/structure/machinery/cm_vending/sorted/attachments/blend/tutorial, attachies_vendor)
+	remove_highlight(attachies_vendor)
+	message_to_player("Success!")
+	stage = TUTORIAL_REQS_LINE_STAGE_GEARBOX
+	var/list/request = list(/obj/item/storage/box/guncase/lmg, /obj/item/explosive/grenade/high_explosive)
+	spawn_agent(request)
+
+/datum/tutorial/marine/reqs_line/proc/continue_stage_mixed()
+	loser_agent = active_agent
+	TUTORIAL_ATOM_FROM_TRACKING(/obj/structure/machinery/cm_vending/sorted/cargo_guns/cargo/blend/tutorial, gear_vendor)
+	remove_highlight(gear_vendor)
+	message_to_player("Success!")
+	stage = TUTORIAL_REQS_LINE_STAGE_MIXED
+	var/list/request = list(/obj/item/attachable/gyro, /obj/item/storage/box/guncase/m41aMK1, /obj/item/explosive/grenade/high_explosive)
+	spawn_agent(request)
+
+/datum/tutorial/marine/reqs_line/proc/loser_got_the_item(turf/source, atom/movable/passer)
+	SIGNAL_HANDLER
+	var/obj/item/prop/replacer/prop = passer
+	if(!istype(prop))
+		return
+	if(prop.original_type != /obj/item/ammo_magazine/rifle/extended)
+		return
+	qdel(prop)
+	loser_agent.say("Nice.")
+	remove_highlight(loser_agent)
+	TUTORIAL_ATOM_FROM_TRACKING(/obj/structure/machinery/cm_vending/sorted/cargo_ammo/cargo/blend/tutorial, ammo_vendor)
+	remove_highlight(ammo_vendor)
+
+	loser_agent = null // Resumes moving
+	// Unstucks the guy at line to move on too if he's already been served
+	if(active_agent)
+		var/list/request = agents[active_agent]
+		if(!length(request))
+			make_agent_leave(TRUE)
+
+/datum/tutorial/marine/reqs_line/proc/continue_stage_survival()
+	message_to_player("Success! You have completed the tutorial!")
+	update_objective("You have finished the tutorial! But there's more if you want to practice.")
+	addtimer(CALLBACK(src, PROC_REF(message_to_player), "You may stay to practice with random orders, or quit with the button at top left."), 3 SECONDS)
```

Comment:
Pressing that button doesn't actually mark the tutorial as completed. Add a different trigger to end the tutorial that ends up calling `end_tutorial(TRUE)`

Reply:
Oh yeah i completely forgot about that, I think ideally this needs a way baked in tutorial code to set completion without ending it

Reply:
https://github.com/cmss13-devs/cmss13/pull/5921